pkg/kds/v2/server/event_based_watchdog.go

```diff
@@ -51,7 +64,7 @@ func (e *EventBasedWatchdog) Start(stop <-chan struct{}) {
 			if err := e.Reconciler.Clear(e.Ctx, e.Node); err != nil {
 				e.Log.Error(err, "reconcile clear failed")
 			}
-			e.Listener.Close()
+			listener.Close()
 			return
 		case <-flushTicker.C:
 			if len(changedTypes) == 0 {
```

Comment:
Would it make sense to have the Reconciler in another goroutine along with `Stop` and resyncs are triggered by an event in the goroutine with the changeTypes ?

Reply:
Yes, but after the reconcile is done, we immediately consume all events that were buffered.

It would help with a situation where we go over a buffer size during the time of reconciliation. Not sure how probable is this with buffer of 100.

I'm ok to change this, but I don't think it's a scope of this PR